File: DAY 13/abstract.cpp
#include<iostream>
using namespace std;
class cryptic{
    public:
    virtual void hash_data(string input)=0;
};
class SHA256_Sim:public cryptic{
    public:
    void hash_data(string input){
        cout<<"SHA256: "<<input<<" parsed to hex blocks.";
    }
};
int main(){
    SHA256_Sim s;
    s.hash_data("Hello");
}

File: DAY 13/abstract.py
from abc import ABC, abstractmethod
class cryptic(ABC):
    @abstractmethod
    def hash_data(self,input):
        pass
class SHA256_Sim(cryptic):
    def hash_data(self,input):
        print(f"SHA256: {input} parsed to hex blocks.")
s=SHA256_Sim()
s.hash_data("hello")